core/services/relay/evm/read/event.go

```diff
@@ -143,7 +143,6 @@ func (b *EventBinding) Bind(ctx context.Context, bindings ...common.Address) err
 
```

Comment:
seems we are still unregistering before registering and also won't be a no-op if the same addresses were provided? 

not an expert here, so I may be way off